### PR TITLE
[wasm] retrieve results from browser bench sample

### PR DESCRIPTION
**Pull request by radekdoulik**
Add simple http server to serve files for the browser-bench sample and let it collect the results at the end.

The results are sent as POST requests and the simple server saves them to the host storage.

### CODE REVIEW

**Review by pavelsavara — APPROVED** [review]

File: src/mono/sample/wasm/simple-server/Program.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace HttpServer
```
Comment (radical) [inline src/mono/sample/wasm/simple-server/Program.cs:8]: use files-scoped namespace

File: src/mono/sample/wasm/simple-server/Program.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
```
Comment (radical) [inline src/mono/sample/wasm/simple-server/Program.cs:7]: Add `#nullable enable`

File: src/mono/sample/wasm/simple-server/Program.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                    break;
+            }
+
+            return 0;
+        }
+
+        private bool StartServer()
+        {
+            var port = 8000 + Random.Shared.Next(1000);
+            var listener = new HttpListener();
```
Comment (radical) [inline src/mono/sample/wasm/simple-server/Program.cs:35]:
```suggestion
            using var listener = new HttpListener();
```

**Review by radical — APPROVED** [review]
I would suggest exploring using aspnetcore here, like we do in https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/BrowserDebugHost/Program.cs .
The suggestions can be addressed in a follow up PR though.

**Pull request merged**